chain-extensions/impls/dapps-staking/src/lib.rs

```diff
@@ -0,0 +1,477 @@
+                        env.write(&res.encode(), false, None).map_err(|_| {
+                            DispatchError::Other("[ChainExtension] DappsStakingExtension SetRewardDestination failed to write result")
+                        })?;
+                        Ok(())
+                    }
+                };
+            }
+
+            DappsStakingFunc::NominationTransfer => {
+                let args: DappsStakingNominationInput<BalanceOf<T>> = env.read_as()?;
+                let origin_smart_contract = Self::decode_smart_contract(args.origin_contract)?;
+                let target_smart_contract = Self::decode_smart_contract(args.target_contract)?;
+                let value: BalanceOf<T> = args.value;
+
+                let base_weight =
+                    <T as pallet_dapps_staking::Config>::WeightInfo::nomination_transfer();
+                env.charge_weight(base_weight)?;
+
+                let caller = env.ext().caller().clone();
+                let call_result = pallet_dapps_staking::Pallet::<T>::nomination_transfer(
+                    RawOrigin::Signed(caller).into(),
+                    origin_smart_contract,
+                    value,
+                    target_smart_contract,
+                );
+                return match call_result {
+                    Err(e) => {
+                        let mapped_error = DSError::try_from(e.error)?;
+                        let res = Result::<(), DSError>::Err(mapped_error);
+                        env.write(&res.encode(), false, None).map_err(|_| {
+                            DispatchError::Other("[ChainExtension] DappsStakingExtension NominationTransfer failed to write result")
+                        })?;
+                        Err(e.error)
+                    }
+                    _ => {
+                        let res = Result::<(), DSError>::Ok(());
+                        env.write(&res.encode(), false, None).map_err(|_| {
+                            DispatchError::Other("[ChainExtension] DappsStakingExtension NominationTransfer failed to write result")
+                        })?;
+                        Ok(())
+                    }
+                };
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl<R> DappsStakingExtension<R> {
+    /// Helper method to decode type SmartContract enum
+    pub fn decode_smart_contract(
+        contract_bytes: [u8; 32],
+    ) -> Result<<R as pallet_dapps_staking::Config>::SmartContract, DispatchError>
+    where
+        R: pallet_dapps_staking::Config,
+        R::AccountId: From<[u8; 32]>,
+    {
+        let account: R::AccountId = contract_bytes.into();
+        // Encode contract address to fit SmartContract enum.
+        // Since the SmartContract enum type can't be accessed from this chain extension,
+        // use locally defined enum clone (see Contract enum)
+        let contract_enum_encoded = Contract::<R::AccountId>::Wasm(account).encode();
```

Comment:
Input address is not necessarily Wasm AccountId. It can be EVM H160.
Is this taken into consideration?